test/ERC20RootVault.test.ts

```diff
@@ -0,0 +1,1149 @@
+                                strategyPerformanceTreasury:
+                                    strategyPerformanceTreasury,
+                                privateVault: true,
+                                managementFee: BigNumber.from(3000),
+                                performanceFee: BigNumber.from(3000),
+                                depositCallbackAddress: lpCallback.address,
+                                withdrawCallbackAddress: lpCallback.address,
+                            } as DelayedStrategyParamsStruct);
+                        await sleep(86400);
+                        await governance
+                            .connect(this.admin)
+                            .commitDelayedStrategyParams(nft);
+
+                        var to = randomAddress();
+                        await expect(
+                            this.subject.withdraw(
+                                to,
+                                BigNumber.from(0),
+                                [BigNumber.from(0), BigNumber.from(0)],
+                                ["0x00", "0x00"]
+                            )
+                        ).to.emit(lpCallback, "WithdrawCallbackCalled");
+                    });
+                });
+            });
+
+            describe("access control:", () => {
+                it("allowed: any address", async () => {
+                    // when all edge cases will be done
+                    await withSigner(randomAddress(), async (signer) => {
+                        await this.subject
+                            .connect(this.admin)
+                            .addDepositorsToAllowlist([this.deployer.address]);
+                        await this.weth
+                            .connect(this.deployer)
+                            .approve(
+                                this.subject.address,
+                                BigNumber.from(10001)
+                            );
+                        await this.usdc
+                            .connect(this.deployer)
+                            .approve(
+                                this.subject.address,
+                                BigNumber.from(10001)
+                            );
+                        await expect(
+                            this.subject
+                                .connect(this.deployer)
+                                .deposit(
+                                    [
+                                        BigNumber.from(10001),
+                                        BigNumber.from(10001),
+                                    ],
+                                    BigNumber.from(0),
+                                    []
+                                )
+                        ).not.to.be.reverted;
+
+                        var to = randomAddress();
+                        await expect(
+                            this.subject.withdraw(
```

Comment:
by default we connect to this.deployer, but we have to make sure that withdraw works for randomAddress(), so pls add connect(signer)